compiler/cli/src/org/jetbrains/kotlin/cli/jvm/compiler/CompileEnvironmentUtil.java

```diff
@@ -131,6 +131,9 @@ private static void copyJarImpl(JarOutputStream stream, File jarPath, boolean re
                 if (e == null) {
                     break;
                 }
+                if ("META-INF/versions/9/module-info.class".equals(e.getName())) {
```

Comment:
Let's just check that the simple name of the file is `module-info.class` via something like this:
```
StringsKt.substringAfterLast(e.getName(), '/', e.getName()).equals("module-info.class")
```
Also, we need a test to ensure that this won't break once we obsolete the JDK 8 support some time in the distant future, and move `module-info.class` from `META-INF` to the jar root.

Reply:
Changes has been made.